### PR TITLE
Add new secret-rotate hook

### PR DESCRIPTION
**Pull request by wallyworld**
Add a new `secret-rotate` hook as well as some new (missing) tests and lint cleanups.

### CODE REVIEW

File: hooks/hooks.go [hunk 1]
```diff
@@ -26,11 +27,13 @@ const (
 	UpdateStatus          Kind = "update-status"
 	PreSeriesUpgrade      Kind = "pre-series-upgrade"
 	PostSeriesUpgrade     Kind = "post-series-upgrade"
+	SecretRotate Kind = "secret-rotate"
```
Comment (ycliuhw) [inline hooks/hooks.go:30]: not sure if it's github showing wrong alignment?

**Review by ycliuhw — APPROVED** [review]
LGTM thanks!

File: hooks/hooks.go (hunk 1 above)
Comment (wallyworld) [reply hooks/hooks.go]: yeah, i just fixed it :-)

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
[add secret hook kind to charm/v8](https://trello.com/c/Ppc0MTNE/1758-add-secret-hook-kind-to-charm-v8)

**Pull request merged**